modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -417,14 +419,20 @@ private static boolean isVoid(Class<?> type) {
     if (!allFields.isEmpty()) {
       return allFields;
     }
-    return getFieldsWithGetterAndSetters(extensionType);
+    return getFieldsWithGetters(extensionType);
   }
 
-  public static Set<Field> getFieldsWithGetterAndSetters(Class<?> extensionType) {
+  public static Set<Field> getFieldsWithGetters(Class<?> extensionType) {
+    return getPropertyDescriptors(extensionType).stream().filter(p -> p.getReadMethod() != null)
+        .map(p -> getField(extensionType, p.getName()))
+        .filter(field -> field != null)
```

Comment:
pretty please, update IntrospectionUtils to return Optional::absent instead of `null`